File: src/Self_defined_Msg/src/main_listener.cpp
# include "ros/ros.h"
# include "person.h"


void callback(const Self_defined_Msg::person::ConstPtr msg){
    ROS_INFO("Recieve the data: %s %d %f",msg->name.c_str(),msg->age, msg->height);
}


int main(int argc, char*argv[]){
    ros::init(argc,argv,"listener");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe<Self_defined_Msg::person>("chat", 10, callback);
    ros::spin();
}

File: src/Self_defined_Msg/scripts/main_listener.py
#! /usr/bin/env python
# coding:utf-8
import rospy
from Self_defined_Msg.msg import person

def callback(msg):
    rospy.loginfo("Recieve the data: %s %d %f",msg.name, msg.age, msg.height)

if __name__ == "__main__":
    rospy.init_node("listener")
    sub = rospy.Subscriber("caht",person,queue_size=10)
    rospy.spin()